lib/codec.js

```diff
@@ -100,6 +122,11 @@
   var longTooBig = f.validator('must be less than 2^53 (javascript safe integer limitation)',
                                f.lessThan(Math.pow(2, 53)));
   var stringOrNullCheck = f.validator('must be a String or null', stringOrNull);
+  var bufferCheck = f.validator('must be a buffer',isBuffer);
+
+  function isBuffer(x){
```

Comment:
is this used only above? If yes I would avoid a new function

Reply:
Not used now but I think it will be used when we will encode basic data types